woocommerce/woocommerce-gutenberg-products-block
# Blockified Order Confirmation

## What

This PR introduces new blocks for the individual parts of the order confirmation page. This includes blocks to display:

- The order status
- The order summary (ID, purchaser, date, total)
- Order details/totals
- Shipping and billing addresses

This replaces the legacy confirmation block which can be found under Appearance > Editor > Templates > Order Confirmation. The switch between legacy and new blocks is a manual process that involves editing the template and clicking the "transform into blocks" button.

Fixes #10021

## Why

This change allows merchants to control the appearance of the order confirmation page. It also modernises how the order confirmation page is rendered.

## Testing Instructions

Perform the following tests as a logged in admin user unless otherwise specified.

1. Go to Appearance > Editor > Templates > Order Confirmation
2. Convert the legacy order confirmation block to individual blocks using the "transform into blocks" button. Confirm the new blocks appear.
3. Focus on each new block and confirm there is an appropriate icon, title, description, and style controls for each. Play around with text colours, background colours, and any other options available to ensure changes apply in the editor.
4. Save changes.
6. Go to the store frontend, add some items to the cart, and place an order.
7. Confirm the confirmation page matches the styling you applied and all blocks are present. You can leave this tab open for testing.
8. Edit the order confirmation page again. Remove some blocks, for example headings.
9. Save changes, then refresh the confirmation page on the frontend. Confirm changes have applied.

#### Downloadable Products

1. Add a downloadable product to your cart and complete checkout.
2. The order confirmation page will be shown—no downloads will be visible. Leave this tab open.
3. In a new tab, go to store admin and edit the order you just placed. Change its status to "processing". This will grant access to downloads.
4. Go back to the confirmation page and refresh.
5. Confirm the download section is now visible.

#### Unauthenticated Views

1. Place an order as a logged in admin user. Leave the confirmation page open.
2. Copy the page address.
3. Open up a new incognito browser window, or better yet, using a different browser. Paste the URL from earlier.
4. Confirm no order details are visible—you should only see some text and a login link as follows:

![Screenshot 2023-09-01 at 13 14 39](https://github.com/woocommerce/woocommerce-blocks/assets/90977/51884f8d-1f63-4cf8-ad49-0b2f392a8335)

5. Close all tabs and log out from your store.
6. Place a new order as a guest. Confirm the order confirmation page shows some order details, but hides email address, payment method, and the full billing/shipping address. This is the limited view that guests can see.
7. Leave the confirmation page open. Copy the page address.
8. Open up a new incognito browser window, or better yet, using a different browser. Paste the URL from earlier.
9. You should still see details if within 10 minutes of placing the order.
10. Wait 10 minutes then refresh the page.
11. Confirm you see a message asking you to either login or fill out the form to see order details as follows:

![Screenshot 2023-09-01 at 13 17 39](https://github.com/woocommerce/woocommerce-blocks/assets/90977/ae660b14-f50c-43a0-8bee-5a608296206c)

12. Enter the incorrect email. See a notice rejecting the request "We were unable to verify the email address you provided. Please try again.".
13. Enter the correct email. See limited order details.
14. Edit the URL of the page to remove the order key. For example, `checkout/order-received/1772/?key=wc_order_GAE4XxQ4k7E77` would become `checkout/order-received/1772`
15. You should no longer be able to provide your email and see order details. The key is required. You will see a generic message only.

![Screenshot 2023-09-01 at 13 53 30](https://github.com/woocommerce/woocommerce-blocks/assets/90977/382a608a-ffc0-4529-976f-f2b58822bab9)

#### Developer tests

Install a 3rd party extension which adds something to the order confirmation page. Confirm that the extension renders it's content on the new order confirmation page. You can find examples on WooComerce.com. One I tried was "Share your purchase" which appends social sharing to the confirmation page.


## Screenshots or screencast

![Screenshot 2023-09-01 at 13 53 50](https://github.com/woocommerce/woocommerce-blocks/assets/90977/81167c90-32c3-4a4b-a82c-e91da89633ab)

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog

> Feature: For users of block based themes, the order confirmation template can now be converted to individual blocks for greater control over layout, styling, and appearance. Go to Appearance > Editor > Templates > Order Confirmation and use the "transform into blocks" button to enable this functionality.
